### PR TITLE
Updates to coin example

### PR DESCRIPTION
**Pull request by ChrisChinchilla**
## Description of the Change

Updates to coin template to add more detail to code comments and ensure code and docs are in sync.

## Github Issues

Part of #170

### CODE REVIEW

File: packages/editor/templates/Coin/contracts/Coin.sol [hunk 1]
```diff
@@ -1,35 +1,56 @@
+// Specifies that the source code is for a version
+// of Solidity greater than 0.5.10
```
Comment (samajammin) [inline packages/editor/templates/Coin/contracts/Coin.sol]: Isn't this specifying greater than or equal to`0.5.0`, not greater than `0.5.10`?

File: packages/editor/templates/Coin/README.md [hunk 2]
```diff
@@ -152,7 +152,7 @@ Coin.prototype.createTokens = function() {
         return;
     }
 
-    this.instance.mint(address, amount, { from: window.web3.eth.accounts[0], gas: 100000, gasPrice: 100000, gasLimit: 100000 },
+    this.instance.mint(address, amount, { from: window.web3.eth.accounts[0], gas: 100000, gasPrice: 100000, gasLimit: 100000 },
```
Comment (samajammin) [inline packages/editor/templates/Coin/README.md]: Small potatoes but I'd remove the trailing space.

File: packages/editor/templates/Coin/README.md [hunk 3]
```diff
@@ -235,7 +235,7 @@ function isValidAddress(address) {
 }
 
 function isValidAmount(amount) {
-    return amount > 0 && typeof Number(amount) == 'number';
+    return amount > 0 && typeof Number(amount) == 'number';
```
Comment (samajammin) [inline packages/editor/templates/Coin/README.md]: Same deal, I'd remove these trailing spaces.

File: packages/editor/templates/Coin/app/app.js [hunk 4]
```diff
@@ -57,34 +59,44 @@ Coin.prototype.showAddressBalance = function(hash, cb) {
...
+    // Validate amount using utility function
     if(!isValidAmount(amount)) {
         console.log("Invalid amount");
         return;
     }
 
     // Transfer amount to other address
+    // Use the public mint function from the smart contract
     this.instance.mint(address, amount, { from: window.web3.eth.accounts[0], gas: 100000, gasPrice: 100000, gasLimit: 100000 }, 
+        // If there's an erro, log it
```
Comment (samajammin) [inline packages/editor/templates/Coin/app/app.js]: `error`

File: packages/editor/templates/Coin/contracts/Coin.sol [hunk 5]
```diff
@@ -1,35 +1,56 @@
...
 
+// A contract is a collection of functions and data (its state)
+// that resides at a specific address on the Ethereum blockchain.
 contract Coin {
-    // The keyword "public" makes variables
-    // accessible from other contracts
+    // The keyword "public" makes variables accessible from outside a contract
+    // and creates a function that other contracts or SDKs can call to access the value
+
+    // An address stores addresses of contracts, or a hash of the public half
```
Comment (samajammin) [inline packages/editor/templates/Coin/contracts/Coin.sol]: Perhaps simpler to say `An address stores addresses of contracts or external (user) accounts`

File: packages/editor/templates/Coin/contracts/Coin.sol [hunk 6]
```diff
@@ -1,35 +1,56 @@
...
         minter = msg.sender;
     }
 
     // Sends an amount of newly created coins to an address
-    // Can only be called by the contract creator
     function mint(address receiver, uint amount) public {
+        // require statements define conditions that must pass
+        // before state is changed
+
+        // Can only be called by the contract creator
```
Comment (samajammin) [inline packages/editor/templates/Coin/contracts/Coin.sol]: Perhaps worth explaining a revert if a `require` fails? Up to you 😄

File: packages/editor/templates/Coin/README.md (hunk 2 above)
Comment (ChrisChinchilla) [reply packages/editor/templates/Coin/README.md]:
```suggestion
    this.instance.mint(address, amount, { from: window.web3.eth.accounts[0], gas: 100000, gasPrice: 100000, gasLimit: 100000 },
```

File: packages/editor/templates/Coin/README.md (hunk 3 above)
Comment (ChrisChinchilla) [reply packages/editor/templates/Coin/README.md]:
```suggestion
    return amount > 0 && typeof Number(amount) == 'number';
```

I have  a dodgy MBP keyboard and get random stuff everywhere right now :(

File: packages/editor/templates/Coin/app/app.js (hunk 4 above)
Comment (ChrisChinchilla) [reply packages/editor/templates/Coin/app/app.js]:
```suggestion
        // If there's an error, log it
```

The keyboard strikes again!

File: packages/editor/templates/Coin/contracts/Coin.sol (hunk 1 above)
Comment (ChrisChinchilla) [reply packages/editor/templates/Coin/contracts/Coin.sol]: I have a vague feeling that someone somewhere told me this isn't the case, but it _feels_ like you're right and I don't know why this  is like this…

```suggestion
// of Solidity greater than 0.5.0
```

File: packages/editor/templates/Coin/contracts/Coin.sol (hunk 5 above)
Comment (ChrisChinchilla) [reply packages/editor/templates/Coin/contracts/Coin.sol]: I was wondering that, but then the statement is only half correct, but at this point, it's probably not something a beginner needs to know.

File: packages/editor/templates/Coin/contracts/Coin.sol (hunk 6 above)
Comment (ChrisChinchilla) [reply packages/editor/templates/Coin/contracts/Coin.sol]: You mean in terms that the state reverts?

File: packages/editor/templates/Coin/contracts/Coin.sol (hunk 1 above)
Comment (samajammin) [reply packages/editor/templates/Coin/contracts/Coin.sol]: Well unless you can cite a source, I'm fairly positive `pragma solidity >=0.5.0` means greater than or equal to `0.5.0`, not greater than `0.5.10`...

File: packages/editor/templates/Coin/contracts/Coin.sol (hunk 5 above)
Comment (samajammin) [reply packages/editor/templates/Coin/contracts/Coin.sol]: Your call but yes, I think we should focus on the level of understanding a beginner needs to know. I'm comfortable with not being completely correct.

File: packages/editor/templates/Coin/contracts/Coin.sol (hunk 6 above)
Comment (samajammin) [reply packages/editor/templates/Coin/contracts/Coin.sol]: Exactly, that when a `require` fails (evaluates to false), it triggers an exception that reverts all modifications done to the state in the current call, including all sub-calls & shows an error to the caller.

**Comment by ChrisChinchilla** [comment]
Think everything is addressed, so being bold and merging

**Pull request merged**